streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -890,17 +888,17 @@ public synchronized ProcessorTopology buildGlobalStateTopology() {
     }
 
     private ProcessorTopology build(final Set<String> nodeGroup) {
-        final List<ProcessorNode> processorNodes = new ArrayList<>(nodeFactories.size());
```

Comment:
Removed the arraylist and use linked hashmap for processorMap to achieve the same thing: following insertion ordering.